tests/unit/amo/components/TestRatingManager.js

```diff
@@ -450,6 +454,80 @@ describe(__filename, () => {
     });
   });
 
+  describe('enableInlineAddonReview', () => {
+    function renderInline(otherProps = {}) {
+      const _config = getFakeConfig({
+        enableInlineAddonReview: true,
+      });
+      return render({ _config, ...otherProps });
+    }
+
+    it('renders a user rating prompt', () => {
+      const root = renderInline();
+
+      expect(root.find(UserRating)).toHaveLength(1);
+      expect(root.find(AddonReviewManager)).toHaveLength(0);
+    });
+
+    it('renders AddonReviewManager after submitting a rating', async () => {
+      const store = createStoreWithLatestReview();
+      const root = renderInline({ store });
+      expect(root.find(AddonReviewManager)).toHaveLength(0);
+
+      await root.instance().onSelectRating(5);
+
+      // Apply setState() changes.
+      expect(root).toHaveState('showTextEntry', true);
+      root.update();
+
+      const manager = root.find(AddonReviewManager);
+      expect(manager).toHaveLength(1);
+      expect(manager).toHaveProp('review', root.instance().props.userReview);
+
+      expect(root.find(UserRating)).toHaveLength(0);
+    });
+
+    it('hides AddonReviewManager when pressing the cancel button', async () => {
+      const root = renderInline({ store: createStoreWithLatestReview() });
+      root.setState({ showTextEntry: true });
+
+      root
+        .find('.RatingManager-cancelTextEntryButton')
+        .simulate('click', createFakeEvent());
+
```

Comment:
`cancelTextEntry` also calls `event.preventDefault()` and we usually verify that in a test.